### PR TITLE
Parse transports in the no extension branch

### PR DESCRIPTION
**Pull request by iuliaion**

### CODE REVIEW

File: u2f-gae-demo/war/js/u2fdemo.js [hunk 1]
```diff
@@ -75,22 +80,10 @@ function hideMessage() {
...
-
-  window.setTimeout(
-    function() {
-      for (i = 0; i < cardChildren.length; i++) {
-        $(cardChildren[i]).removeClass("highlight", 2000);
-      }
-    },
-    500
-  );
+  var cardContent = $("#" + token.public_key).find(".cardContent");
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2fdemo.js]: Can we revert this change?  It's awkward, but needed because iOS can't handle the jQuery for some reason.

File: u2f-ref-code/java/src/com/google/u2f/server/impl/U2FServerReferenceImpl.java [hunk 2]
```diff
@@ -309,8 +409,7 @@ private void verifyBrowserData(JsonElement browserDataAsElement,
   private void verifyOrigin(String origin) throws U2FException {
     if (!allowedOrigins.contains(canonicalizeOrigin(origin))) {
       throw new U2FException(origin +
-          " is not a recognized home origin for this backend" +
-    		  Joiner.on(", ").join(allowedOrigins));
+          " is not a recognized home origin for this backend");
```
Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/U2FServerReferenceImpl.java]: why this change?

File: u2f-gae-demo/war/js/u2fdemo.js [hunk 3]
```diff
@@ -214,40 +207,3 @@ function onError(code, enrolling) {
     break;
   }
 }
-
-if (navigator.userAgent.indexOf("iPhone") > -1) {
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2fdemo.js]: can you leave this code in for now so Kaiyu can test and then remove it when it goes into the u2f-api.js file?

File: u2f-gae-demo/war/js/u2fdemo.js (hunk 1 above)
Comment (iuliaion) [inline u2f-gae-demo/war/js/u2fdemo.js]: Done. Sorry, I though I had fixed this.

File: u2f-gae-demo/war/js/u2fdemo.js (hunk 3 above)
Comment (iuliaion) [inline u2f-gae-demo/war/js/u2fdemo.js]: Done

File: u2f-ref-code/java/src/com/google/u2f/server/impl/U2FServerReferenceImpl.java (hunk 2 above)
Comment (iuliaion) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/U2FServerReferenceImpl.java]: The master branch differs here. I reverted.

Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/U2FServerReferenceImpl.java]: Seems like this is a nice to have, maybe also add it to master branch?

File: u2f-gae-demo/war/js/u2fdemo.js (hunk 1 above)
Comment (leshi) [inline u2f-gae-demo/war/js/u2fdemo.js]: I still see the change.  Are you sure you reverted?

File: u2f-gae-demo/war/js/u2fdemo.js (hunk 3 above)
Comment (leshi) [inline u2f-gae-demo/war/js/u2fdemo.js]: Also don't see the change.  Maybe you didn't push?

**Comment by leshi** [comment]
LGTM
